src/task.js

```diff
@@ -0,0 +1,110 @@
+// This a class containing all the necessary operations
+import dots from './images/menu.png';
+import unchecked from './images/unchecked.png';
+import mark from './images/check.png';
+import recycle from './images/bin.png';
+
+const ul = document.getElementById('to-do-list');
+export default class TaskHandling {
+  constructor(tasks) {
+    this.tasks = tasks;
+    this.taskCount = tasks.length;
+    ul.innerHTML = '';
+    this.populate();
+  }
+
+  add(ele) {
+    if (ele.value === '') {
+      return;
+    }
+    this.taskCount += 1;
+    this.tasks.push({ index: this.taskCount, desc: ele.value, completed: false });
+    localStorage.setItem('listOfTasks', JSON.stringify(this.tasks));
+    ul.innerHTML = '';
+    ele.value = '';
+    this.populate();
+  }
+
+  remove(ele) {
+    this.tasks = this.tasks.filter((e) => e.index !== +ele.parentElement.id);
+    for (let i = 0; i < this.tasks.length; i += 1) {
+      this.tasks[i].index = i + 1;
+    }
+    localStorage.setItem('listOfTasks', JSON.stringify(this.tasks));
+    ul.removeChild(ele.parentElement);
+  }
```

Comment:
- [ ] Kindly make sure a task is deleted when the delete icon is clicked

Reply:
Actually based on the design example given to us, you can only delete the task when it's empty.
But I will implement what you said. Thank you.